fn/io_test.go

```diff
@@ -35,6 +36,11 @@ func TestWriteFile(t *testing.T) {
 	err = os.Chmod(filename, 0444)
 	require.NoError(t, err, "couldn't change file to read-only")
 
+	// Root can bypass read-only file mode.
+	if syscall.Geteuid() == 0 {
```

Comment:
can you add this check in the beginning of the test and just skip it if running as root ?